### DISCUSSION TITLE
How to handle dependencies on custom webpack loaders?

### DISCUSSION THREAD

**Discussion by pongells** [yarnpkg/berry#2977] [Q&A] [answered]
Hello all,

In my application webpack config I have the following `example-loader`:

```
const config: webpack.Configuration = {
  plugins: [
   ...
  ],
  module: {
    ...
  },
  resolveLoader: {
    alias: {
      'example-loader': path.join(__dirname, 'webpack-example-loader.js'),
    },
  },
};
```

which when called exports an object like:

```
return `module.exports = {
  fileName: ...,
  contentFormatted: ...
}`;
```

and I am using in the code as follows:

```
const myExample = require('!!example-loader!./examples/something/my-example.html'),
``` 

I am not 100% this is the best way to go around it, but it works. 

The problem is, using `@yarnpkg/doctor` I am getting the following:

```
➤ YN0000: │ C:/.../file.ts:105:9: Undeclared dependency on example-loader
```

So how am I supposed to tell Yarn that `example-loader` is an internal dependency on a webpack loader? 

Thanks

**Comment by arcanis** [comment, marked as answer]
You can't - it's part of why the Doctor is more a zero-config quick CLI check than a CI-ready tool, since it's meant to report things that are *probably* errors, even if it means being overly cautious sometimes. You'll always have false positives like in cases like this.

**Reply by pongells** [reply]
I see, thanks!

**Reply by pongells** [reply]
@arcanis you think it would make sense to implement a white-list for such cases? I just figured out we could modify the `isValidDependency()` method to check against such list.. 

```
function isValidDependency(ident, { workspace }) {
    if (ident.identHash === workspace.locator.identHash)
        return true;
    if (workspace.manifest.hasDependency(ident))
        return true;
    // --- added ---
    if (exclusions.includes(ident.name))
        return true;
    return false;
}
```
